### DISCUSSION TITLE
Errors when trying to install hid-fanatecff

### DISCUSSION THREAD

**Discussion by Paradodajka** [gotzl/hid-fanatecff#88] [Q&A] [answered]
I did install all dependencies but after running command make install I always get this (currently on latest stable Steam Deck channel):

:: Compiling Fanatec kernel module
========================================
make -C /lib/modules/`uname -r`/build M=$PWD
make[1]: Entering directory '/usr/lib/modules/6.1.52-valve16-1-neptune-61/build'
make[1]: *** No targets specified and no makefile found.  Stop.
make[1]: Leaving directory '/usr/lib/modules/6.1.52-valve16-1-neptune-61/build'
make: *** [Makefile:8: default] Error 2
[sudo] password for deck: 

:: Installing Fanatec kernel module/udev rule
=====================================================
cp: cannot stat 'hid-fanatec.ko': No such file or directory
make: *** [Makefile:18: install] Error 1

Any help please?

**Comment by gotzl** [comment, marked as answer]
Hmm.. Do you have the kernel headers installed? S.t like `sudo pacman -S linux-neptune-61 linux-neptune-61-headers` ?

**Reply by Paradodajka** [reply]
I tried to reinstall those but it looks like there is a missing file "xhci_pci"

warning: linux-neptune-61-6.1.52.valve16-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (2) linux-neptune-61-6.1.52.valve16-1  linux-neptune-61-headers-6.1.52.valve16-1

Total Download Size:   148.29 MiB
Total Installed Size:  259.00 MiB
Net Upgrade Size:      144.50 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
 linux-neptune-61-headers-6...    33.3 MiB  16.9 MiB/s 00:02 [#################################] 100%
 linux-neptune-61-6.1.52.va...   115.0 MiB  38.1 MiB/s 00:03 [#################################] 100%
 Total (2/2)                     148.3 MiB  40.2 MiB/s 00:04 [#################################] 100%
(2/2) checking keys in keyring                               [#################################] 100%
(2/2) checking package integrity                             [#################################] 100%
(2/2) loading package files                                  [#################################] 100%
(2/2) checking for file conflicts                            [#################################] 100%
(2/2) checking available disk space                          [#################################] 100%
:: Processing package changes...
(1/2) reinstalling linux-neptune-61                          [#################################] 100%
(2/2) installing linux-neptune-61-headers                    [#################################] 100%
:: Running post-transaction hooks...
(1/5) Arming ConditionNeedsUpdate...
(2/5) Updating module dependencies...
(3/5) Installing linux-neptune-splash screen to EFI...
(4/5) Updating linux initcpios...
==> Building image from preset: /etc/mkinitcpio.d/linux-neptune-61.preset: 'default'
==> Using default configuration file: '/etc/mkinitcpio.conf'
  -> -k /boot/vmlinuz-linux-neptune-61 -g /boot/initramfs-linux-neptune-61.img --microcode /boot/amd-ucode.img
==> Using drop-in configuration file: '20-steamdeck.conf'
==> Starting build: '6.1.52-valve16-1-neptune-61'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [steamos]
==> ERROR: file not found: ''
  -> Running build hook: [steam-deck]
==> WARNING: Possibly missing firmware for module: 'xhci_pci'
  -> Running build hook: [modconf]
  -> Running build hook: [keyboard]
  -> Running build hook: [block]
  -> Running build hook: [filesystems]
  -> Running build hook: [resume]
==> Generating module dependencies
==> Creating zstd-compressed initcpio image: '/boot/initramfs-linux-neptune-61.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
==> Building image from preset: /etc/mkinitcpio.d/linux-neptune-61.preset: 'fallback'
==> Using default configuration file: '/etc/mkinitcpio.conf'
  -> -k /boot/vmlinuz-linux-neptune-61 -g /boot/initramfs-linux-neptune-61-fallback.img -S autodetect --microcode /boot/amd-ucode.img
==> Using drop-in configuration file: '20-steamdeck.conf'
==> Starting build: '6.1.52-valve16-1-neptune-61'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [steamos]
==> ERROR: file not found: ''
  -> Running build hook: [steam-deck]
==> WARNING: Possibly missing firmware for module: 'xhci_pci'
  -> Running build hook: [modconf]
  -> Running build hook: [keyboard]
  -> Running build hook: [block]
  -> Running build hook: [filesystems]
  -> Running build hook: [resume]
==> Generating module dependencies
==> Creating zstd-compressed initcpio image: '/boot/initramfs-linux-neptune-61-fallback.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
error: command failed to execute correctly
(5/5) Managing kdump minimal initramfs
==> ERROR: file not found: ''
==> WARNING: Possibly missing firmware for module: 'xhci_pci'
==> WARNING: errors were encountered during the build. The image may not be complete.
error: command failed to execute correctly

**Reply by gotzl** [reply]
Don't know exactly why you get this warning/error. Possible you can ignore this. It seems however that the important bits got installed. Does it compile now?

**Reply by Paradodajka** [reply]
Yes, it works now! Thank you So much
